### ISSUE TITLE
Calendar Share Functionality missing

### ISSUE THREAD

**Issue by kumzugloom** [nextcloud/server#866]
### Steps to reproduce
When I create calendars with one user, I'd like to share them with another user or even publicly. This was possible in OC9, however, I cannot find the button in the "edit calendar" section, I can only change color and name there.

### Expected behaviour
Users should be able to share their calendars with other users.
(As described in https://nextcloud.com/features/#calendarcontacts:  "Private and shared calendars and addressbooks with permissions management")

### Actual behaviour
Users cannot share their calendars.

### Server configuration
**Operating system**:
Raspbian

**Web server:**
Nginx

**Database:**
SQLite

**PHP version:**
PHP 5.6.24-0+deb8u1

**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 9.0.50 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh install

**Where did you install Nextcloud from:**
Nextcloud.com

**Comment by mar1u5** [comment]
Seems to be a calendar-app issue, which has nothing to do with the Nextcloud Server. Could you provide me a screenshot :grin:

**Comment by kumzugloom** [comment]
Hi @mar1u5, 
thanks for your quick reply. Yes, it seems to be a calendar app issue, but i did not found the nextcloud calendar app bugtracker. Therefore I thought it might be maintained in here as well.

Here you have the screenshot not showing the dialog:
![nextcloud_cal](https://cloud.githubusercontent.com/assets/1199617/17656395/f5671a84-62b9-11e6-9e8e-d8dbd9afd326.png)

I have Thunderbird Lightning on Windows and DAVDroid on Android as Clients. Syncronizing works fine, but I cannot change permissions of calendars in those clients.

I created this Raspberry completely from scratch, so there might be slight differences in the configuration, but currently I don't know them. I'll try to investigate further.

**Comment by mar1u5** [comment]
THX @Pommesumdreher 👊 

I can reproduce this when I activate AdBlock Plus 😈  Try whether this solves the problem for you - and report back @kumzugloom ;)

... Contacts / Calendar and Mail repositories (with the bugtracker) are _currently_ in ownCloud :wink:
_Thats the confusion I've talked about to you @jancborchardt @karlitschek_

**I'll close this issue here for now, because this is the `server` and not the `calendar` 😁
If you are still experiencing problems with the calendar - please open a new issue here:**
https://github.com/owncloud/calendar

The new issue you open in `owncloud/calendar` should contain the link to this issue as a reference, and please mention me with "@mar1u5" too, so that I am informed about the status of your issue 😼

**Comment by kumzugloom** [comment]
Thanks a lot @Pommesumdreher and @mar1u5!
That was exactly the problem. Thank you also for pointing me to the right direction regarding (potential) bug reports in the future. :+1:

**Comment by jancborchardt** [comment]
@mar1u5 @Pommesumdreher @kumzugloom does anyone of you want to take initiative on contacting AdBlock Plus to not block this? I think it has to do with the icon being named »share«. We need to have them add an exception so it’s not blocked in Nextcloud, especially as the audience of Nextcloud and Adblockers probably have quite some overlap.

**Comment by mar1u5** [comment]
@jospoortvliet ... That could be a Job for you.

**Comment by jancborchardt** [comment]
@mar1u5 @Pommesumdreher @kumzugloom or for you folks since you encounter the issue. ;) Remember Nextcloud is community-driven, and especially this shortly before the conf, Jos will have a bunch of other things to do. :)

**Comment by kumzugloom** [comment]
Hi @mar1u5 @Pommesumdreher @jancborchardt,
you are probably correct. I just searched the relevant forums and was able to identify, that this issue was raised to AdBlock and EasyList. However, the response was (in short): "Won't fix, just rename your button."

Below you will find the links to the forums:
AdBlock Plus: https://adblockplus.org/forum/viewtopic.php?f=18&t=44184&p=148242&hilit=owncloud#p148242
Easylist: https://forums.lanik.us/viewtopic.php?f=64&t=28785

**Comment by jancborchardt** [comment]
Yeah, so the important one is this: https://forums.lanik.us/viewtopic.php?f=64&t=28785 – mind commenting if there can be a rule that Nextcloud/ownCloud are exempt from that icon blocking? Cause icon-share is also used by other frameworks and is kinda stupid to block for everyone.

**Comment by Bullnados** [comment]
Some news here?

**Comment by Pommesumdreher** [comment]
Since Adblock added their own Ads, I switched to uBlock Origin. It works better for me and there are no problems with nextcloud ;)

**Comment by Bullnados** [comment]
I'm also using uBlock Origin and i still have the problem

**Comment by Pommesumdreher** [comment]
Which Version of uBlock Origin and Nextcloud do you use?

**Comment by Bullnados** [comment]
NC 11.0.2 and origin 1.11.4

**Comment by Pommesumdreher** [comment]
Which browser do you use? I am using firefox (I have to leave my computer now, so I will answer a bit later)

**Comment by Bullnados** [comment]
ff 53 beta x64
Perhaps it is some of the block lists

**Comment by Pommesumdreher** [comment]
Yes, maybe. Do you have a Number in the Sign of uBlock Origin on the top right, when Nextcloud is open?

**Comment by Pommesumdreher** [comment]
If yes, what is listed under "blocked requests"?

**Comment by GetBoz** [comment]
In uBlock Origin, the share icon is hidden by the ##.icon-share filter of Fanboy's Lists

"Static filter **##.icon-share** found in :"
![fanboy](https://cloud.githubusercontent.com/assets/8277114/24508752/bc5f6922-1564-11e7-939c-4a37307f10c0.PNG)

**Comment by jancborchardt** [comment]
All fixed now! :) Please review https://github.com/nextcloud/server/pull/4337 and also the pull requests to Calendar https://github.com/nextcloud/calendar/pull/417 and Contacts https://github.com/nextcloud/contacts/pull/188 :tada:

**Comment by sebalis** [comment]
The share icons for unshared calendar items are still blocked by the “Fanboy’s Social” list on uBlock Origin in Nextcloud 17.0.3 (official Calendar app). No surprise because they use the class “icon-share”, not “icon-shared” as for unshared files or, indeed, shared calendars (so the share icons for all of these are not blocked).

**Comment by mc256** [comment]
ad block plus is also blocking this icon
